### PR TITLE
Add CBMC proof of initFileHandler 

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of initFileHandler function.

Description
-----------
This PR adds the files required for the CBMC proof of initFileHandler function. 

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gshvang — APPROVED** [review]

File: test/cbmc/source/stubs.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * @file stubs.c
+ * @brief Includes the definition of all the stubs required for the CBMC proofs.
+ */
+/*  Ota Agent includes. */
+#include "ota.h"
+
+OtaErr_t initFileTransferStub( OtaAgentContext_t * agent )
+{
+    OtaErr_t err;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:32]: I think err should be initailized here.

File: test/cbmc/source/stubs.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    return err;
+}
+
+OtaOsStatus_t startTimerStub( OtaTimerId_t otaTimerId,
+                              const char * const pTimerName,
+                              const uint32_t timeout,
+                              OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:47]: I think status should be initailized here.

File: test/cbmc/source/stubs.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    __CPROVER_assert( pTimerName != NULL, "Error: TimerName cannot be NULL" );
+
+    __CPROVER_assert( callback != NULL, "Error: callback should be pointing to a function." );
+
+    return status;
+}
+
+OtaOsStatus_t stopTimerStub( OtaTimerId_t otaTimerId )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:61]: I think status should be initailized here.

**Review by gshvang — APPROVED** [review]

**Review by gshvang — DISMISSED** [review]

File: test/cbmc/proofs/initFileHandler/initFileHandler_harness.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+extern OtaDataInterface_t otaDataInterface;
+extern OtaErr_t initFileHandler( const OtaEventData_t * pEventData );
+
+void initFileHandler_harness()
+{
+    OtaErr_t err;
+    OtaEventData_t * pEventData;
+    OtaInterfaces_t otaInterface;
+
+    pEventData = ( OtaEventData_t * ) malloc( sizeof( OtaEventData_t ) );
```
Comment (paulbartell) [inline test/cbmc/proofs/initFileHandler/initFileHandler_harness.c:43]: Does this need to be heap allocated?

File: test/cbmc/proofs/initFileHandler/initFileHandler_harness.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    otaDataInterface.initFileTransfer = initFileTransferStub;
+    otaInterface.os.timer.start = startTimerStub;
+    otaInterface.os.timer.stop = stopTimerStub;
```
Comment (paulbartell) [inline test/cbmc/proofs/initFileHandler/initFileHandler_harness.c:47]: The remainder of the otaAgent object should be unconstrained.

File: test/cbmc/proofs/initFileHandler/initFileHandler_harness.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+     * during the initialization of OTA specifically in the OTA_Init function. */
+    otaAgent.pOtaInterface = &otaInterface;
+
+    err = initFileHandler( pEventData );
+
+    /* initFileHandler returns the values which follow OtaErr_t enum. If it does not, then
+     * there is a problem. */
+    __CPROVER_assert( ( err >= OtaErrNone ) && ( err <= OtaErrActivateFailed ),
+                      "Invalid return value from initFileHandler: Expected a value from OtaErr_t enum." );
+
```
Comment (paulbartell) [inline test/cbmc/proofs/initFileHandler/initFileHandler_harness.c:59]: the resulting otaAgent.requestMomentum field is not validated.

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**